src/test/scala/zio/prelude/IdempotentSpec.scala

```diff
@@ -9,6 +11,23 @@ object IdempotentSpec extends DefaultRunnableSpec {
 
   val anyMaxInt: Gen[Random, Max[Int]] = Gen.anyInt.map(Max(_))
 
+  val anyPartialOrdering: Gen[Random, PartialOrdering] = Gen.anyInt.map { n =>
+    abs(n) % 4 match {
+      case 0 => Ordering.LessThan
+      case 1 => Ordering.Equals
+      case 2 => Ordering.GreaterThan
+      case 3 => PartialOrdering.NoOrder
+    }
+  }
+
+  val anyOrdering: Gen[Random, Ordering] = Gen.anyInt.map { n =>
+    abs(n) % 3 match {
+      case 0 => Ordering.LessThan
+      case 1 => Ordering.Equals
+      case 2 => Ordering.GreaterThan
+    }
+  }
+
```

Comment:
I haven't personally encountered this pattern before for property testing, but I like it.